interface/forms/LBF/printable.php

```diff
@@ -12,73 +12,155 @@
-if (!empty($jobj['aco'])) {
-    $LBF_ACO = explode('|', $jobj['aco']);
-}
+// True if to display as a form to complete, false to display as information.
+$isform = empty($_REQUEST['isform']) ? 0 : 1;
+
+$CPR = 4; // cells per row
 
+$grparr = array();
+getLayoutProperties($formname, $grparr, '*');
+$lobj = $grparr[''];
+$formtitle = $lobj['grp_title'];
+if (!empty($lobj['grp_columns'])) $CPR = intval($lobj['grp_columns']);
+if (!empty($lobj['grp_size'   ])) $FONTSIZE = intval($lobj['grp_size']);
+if ($lobj['grp_services']) $LBF_SERVICES_SECTION = $lobj['grp_services'] == '*' ? '' : $lobj['grp_services'];
+if ($lobj['grp_products']) $LBF_PRODUCTS_SECTION = $lobj['grp_products'] == '*' ? '' : $lobj['grp_products'];
+if ($lobj['grp_diags'   ]) $LBF_DIAGS_SECTION    = $lobj['grp_diags'   ] == '*' ? '' : $lobj['grp_diags'   ];
+
+// Check access control.
+if (!empty($lobj['aco_spec'])) $LBF_ACO = explode('|', $lobj['aco_spec']);
 if (!acl_check('admin', 'super') && !empty($LBF_ACO)) {
     if (!acl_check($LBF_ACO[0], $LBF_ACO[1])) {
         die(xlt('Access denied'));
     }
 }
 
+// Html2pdf fails to generate checked checkboxes properly, so write plain HTML
+// if we are doing a visit-specific form to be completed.
+$PDF_OUTPUT = ($formid && $isform) ? false : true;
+// $PDF_OUTPUT = false; // debugging
+
+if ($PDF_OUTPUT) {
+    require_once("$srcdir/html2pdf/vendor/autoload.php");
+    $pdf = new HTML2PDF('P', 'Letter', 'en');
+    $pdf->setTestTdInOnePage(false); // Turn off error message for TD contents too big.
+    $pdf->pdf->SetDisplayMode('real');
+    ob_start();
+}
+
+if ($visitid && (isset($LBF_SERVICES_SECTION) || isset($LBF_DIAGS_SECTION) || isset($LBF_PRODUCTS_SECTION))) {
+    require_once("$srcdir/FeeSheetHtml.class.php");
+    $fs = new FeeSheetHtml($pid, $encounter);
+}
+
 $fres = sqlStatement("SELECT * FROM layout_options " .
   "WHERE form_id = ? AND uor > 0 " .
-  "ORDER BY group_name, seq", array($formname));
+  "ORDER BY group_id, seq", array($formname));
 ?>
+<?php if (!$PDF_OUTPUT) { ?>
 <html>
 <head>
 <?php html_header_show();?>
+<?php } ?>
 
 <style>
+
+<?php if ($PDF_OUTPUT) { ?>
+td {
+ font-family: Arial;
+ font-weight: normal;
+ font-size: <?php echo $FONTSIZE; ?>pt;
+}
+<?php } else { ?>
 body, td {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: normal;
- font-size: 9pt;
+ font-size: <?php echo $FONTSIZE; ?>pt;
 }
-
 body {
  padding: 5pt 5pt 5pt 5pt;
 }
+<?php } ?>
+
+p.grpheader {
+ font-family: Arial;
+ font-weight: bold;
+ font-size: <?php echo round($FONTSIZE * 1.00); ?>pt;
+ margin-bottom: <?php echo round($FONTSIZE * 0.00); ?>pt;
+ margin-top: <?php echo round($FONTSIZE * 0.44); ?>pt;
+}
+
+div.grpheader {
+ font-family: Arial;
+ font-weight: bold;
+ font-size: <?php echo round($FONTSIZE * 1.00); ?>pt;
+ color: #000000;
+ background-color: #cccccc;
+ padding: 2pt;
+ margin-bottom: <?php echo round($FONTSIZE * 0.22); ?>pt;
+ margin-top: <?php echo round($FONTSIZE * 0.22); ?>pt;
```

Comment:
text() around all $FONTSIZE above

Reply:
round() will always return a float so it's not necessary in that case.